main/src/ca/uwaterloo/flix/language/phase/OccurrenceAnalyzer1.scala

```diff
@@ -0,0 +1,580 @@
+    */
+  private def visitExp(sym0: Symbol.DefnSym, exp00: MonoAst.Expr): (OccurrenceAst1.Expr, OccurInfo) = {
+
+    /**
+      * Local visitor that captures `sym0` since it never changes.
+      * The implicit parameter `letBinding` is `Some(sym)` if visiting the right-hand side
+      * of a let-binding, i.e., in `let sym = exp` it recurses on `exp` with `letBinding = Some(sym)`.
+      */
+    def visit(exp0: MonoAst.Expr)(implicit letBinding: Option[VarSym]): (OccurrenceAst1.Expr, OccurInfo) = exp0 match {
+      case MonoAst.Expr.Cst(cst, tpe, loc) =>
+        (OccurrenceAst1.Expr.Cst(cst, tpe, loc), OccurInfo.One)
+
+      case MonoAst.Expr.Var(sym, tpe, loc) =>
+        (OccurrenceAst1.Expr.Var(sym, tpe, loc), OccurInfo.One + (sym -> Once))
+
+      case MonoAst.Expr.Lambda(fparam, exp, tpe, loc) =>
+        val fp = visitFormalParam(fparam)
+        val (e, o) = visit(exp)
+        val o1 = captureVars(o)
+        val occur = o1.get(fp.sym)
+        val o2 = o1 - fp.sym
+        (OccurrenceAst1.Expr.Lambda((fp, occur), e, tpe, loc), increment(o2))
+
+      case MonoAst.Expr.ApplyAtomic(op, exps, tpe, eff, loc) =>
+        val (es, o) = visitExps(exps)
+        val o1 = visitAtomicOp(op, o)
+        (OccurrenceAst1.Expr.ApplyAtomic(op, es, tpe, eff, loc), increment(o1))
+
+      case MonoAst.Expr.ApplyClo(exp, exps, tpe, eff, loc) =>
+        val (e, o1) = visit(exp)
+        val (es, o2) = visitExps(exps)
+        val o3 = combineInfo(o1, o2)
+        val o4 = combineApplyCloInfo(o3, e)
+        (OccurrenceAst1.Expr.ApplyClo(e, es, tpe, eff, loc), increment(o4))
+
+      case MonoAst.Expr.ApplyDef(sym, exps, itpe, tpe, eff, loc) =>
+        val (es, o1) = visitExps(exps)
+        val o2 = o1 :+ sym -> Once
+        (OccurrenceAst1.Expr.ApplyDef(sym, es, itpe, tpe, eff, loc), increment(o2))
+
+      case MonoAst.Expr.ApplyLocalDef(sym, exps, tpe, eff, loc) =>
+        val (es, o1) = visitExps(exps)
+        val o2 = o1 + (sym -> Once)
+        (OccurrenceAst1.Expr.ApplyLocalDef(sym, es, tpe, eff, loc), increment(o2))
+
+      case MonoAst.Expr.Let(sym, exp1, exp2, tpe, eff, loc) =>
+        val (e1, o1) = visit(exp1)(Some(sym))
+        val (e2, o2) = visit(exp2)
+        val o3 = combineInfo(o1, o2)
+        val occur = o3.get(sym)
+        val o4 = o3 - sym
+        (OccurrenceAst1.Expr.Let(sym, e1, e2, tpe, eff, occur, loc), increment(o4))
+
+      case MonoAst.Expr.LocalDef(sym, fparams, exp1, exp2, tpe, eff, loc) =>
+        val fps = fparams.map(visitFormalParam)
+        val (e1, o10) = visit(exp1)
+        val o1 = captureVars(o10)
+        val (e2, o2) = visit(exp2)
+        val o3 = combineInfo(o1, o2)
+        val occur = o3.get(sym)
+        val o4 = o3 - sym
+        (OccurrenceAst1.Expr.LocalDef(sym, fps, e1, e2, tpe, eff, occur, loc), increment(o4))
+
+      case MonoAst.Expr.Scope(sym, rvar, exp, tpe, eff, loc) =>
+        val (e, o1) = visit(exp)
+        val o2 = o1 :+ sym0 -> DontInline
```

Comment:
why are functions with scope not inlined?

Reply:
Leftover from old inliner